apps/prairielearn/src/server.js

```diff
@@ -1827,6 +1827,37 @@ module.exports.initExpress = function () {
     require('./pages/publicQuestions/publicQuestions'),
   ]);
 
+  // Client files for questions
+  app.use('/pl/public/course/:course_id/question/:question_id/clientFilesCourse', [
+    function (req, res, next) {
+      res.locals.public = true;
+      next();
+    },
```

Comment:
What is the recommended way to pass a parameter from the main server file down the the handler? I'm almost positive that attaching to locals is _not_ it, but I wasn't sure what the preferred method was.

Reply:
A couple options here:

- Export a function that takes an argument and returns the middleware. We use this pattern with the `logPageView` middleware:
  https://github.com/PrairieLearn/PrairieLearn/blob/9fe3e1346c472ed60d0479083e564fbcc7aa293e/apps/prairielearn/src/server.js#L1415

- Export multiple middlewares from the same file, e.g. `clientFilesQuestion` and `clientFilesQuestionPublic`.

Reply:
ok, I figured something like this would be nice, I didn't realize we had examples. Done.